### PR TITLE
Blog - Comparing Local Kubernetes Development Solutions

### PR DESCRIPTION
**Pull request by msali123**
Commit 2e4e7b024d0e19c428cbc853028f7dd5df4ad1f7 deployed to https://62aa1a5e98b4475cdb083beb--ephemeral-narwhal-3ac3b2.netlify.app/

### CODE REVIEW

File: blog/_posts/2022-06-21-k8s-dev-solutions.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+The most important thing to look into when deciding on a local Kubernetes solution is what platforms it supports. If you're using Windows as your host system, there's no point in looking at a tool that only supports Linux. You also need to understand whether the tool runs natively on the host, in containers, in VMs, or in a completely different way.
+
+In addition, it's important to understand whether you're getting a full-blown Kubernetes version on a small scale or if you're getting a modified version. And it's important to understand the performance implications of the tool you're choosing.
+
+To understand how performant these solutions are, the [Online Boutique](https://github.com/GoogleCloudPlatform/microservices-demo.git), a cloud-native microservices demo app by Google, will be deployed on each solution and timed. Remember that this is purely an indication of performance, with a sample size of one. This is also the reason why deploy times won't be presented in precise measurements.
+
+### Minikube
+
+![minikube logo]({{site.images}}{{page.slug}}/E51VHlu.png)
```
Comment (jalletto) [inline blog/_posts/2022-06-21-k8s-dev-solutions.md:63]:
```suggestion
![minikube logo]({{site.images}}{{page.slug}}/E51VHlu.png)\
```

**Review by jalletto — CHANGES_REQUESTED** [review]
This is looking really great! I have a couple things. 

1. For images where the caption is just a description of the image like "Github Logo", you can add a`\` at the end. This keeps the caption as meta data, but it won't be visible when the site is deployed. For captions that actually offer something new like "Here you can see the output in the terminal" It's ok to leave the caption as is. Hopefully that makes sense.

2. Can you move the logos up to the section Overview of Each Tool?

3. Can you find generic images for the sections Complexity of Setup, Flexibility, and Support?

File: blog/_posts/2022-06-21-k8s-dev-solutions.md (hunk 1 above)
Comment (msali123) [reply blog/_posts/2022-06-21-k8s-dev-solutions.md:63]: Done

File: blog/_posts/2022-06-21-k8s-dev-solutions.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+minikube has support for all three major operating systems: Windows, macOS, and Linux. This means you likely don't have to worry if you plan on rolling out minikube organization-wide since pretty much any PC is able to run it. On top of that, you also get great platform support in terms of *how* minikube should be run, given that it supports many different [drivers](https://minikube.sigs.k8s.io/docs/drivers/), like Docker, [kvm2](https://minikube.sigs.k8s.io/docs/drivers/kvm2/), and [VirtualBox](https://www.virtualbox.org).
+
+With minikube, you're getting pretty close to running inside a full Kubernetes cluster. You'll still feel like you're running a local cluster at times, but more on that later.
+
+In terms of performance, minikube does fairly well, spinning up the [microservice demo](https://github.com/GoogleCloudPlatform/microservices-demo.git) in just a few minutes.
+
+### Kind
+
+![Kind logo]({{site.images}}{{page.slug}}/9eE2wL1.png)
```
[the remark this answers is not in the document]
Comment (msali123) [reply blog/_posts/2022-06-21-k8s-dev-solutions.md:73]: Done

File: blog/_posts/2022-06-21-k8s-dev-solutions.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+Like minikube, kind supports all three major platforms. It works by spinning up Docker containers to act as nodes in your cluster, which are based on an image [created by kind](https://kind.sigs.k8s.io/docs/user/quick-start/#creating-a-cluster). However, if you want to use your own image, that's also possible using the `--image` flag.
+
+Because the image is developed by kind themselves, there may be variations compared to what you find in [Azure](/blog/azure-functions-node) Kubernetes Service (AKS) or Google Kubernetes Engine (GKE), but in the end, it's a solution that conforms to Kubernetes conventions, so you should be able to run all your regular workloads using kind. You'll also get great performance, which is evident with the [microservice demo](https://github.com/GoogleCloudPlatform/microservices-demo.git) deploying in just about a minute and a half.
+
+### K3s
+
+![K3s logo]({{site.images}}{{page.slug}}/NfeQXkM.png)
```
[the remark this answers is not in the document]
Comment (msali123) [reply blog/_posts/2022-06-21-k8s-dev-solutions.md:81]: Done

File: blog/_posts/2022-06-21-k8s-dev-solutions.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+K3s is the first tool on this list that only supports running on Linux due to the fact that K3s isn't actually made to *be* a development solution. Rather, it was developed as a low-resource alternative to Kubernetes (hence the name K3s, which is a play on the abbreviation K8s).
+
+This means that you don't install K3s as a tool with `brew` or `choco`, rather you install it as a [Linux service](https://www.hostinger.com/tutorials/manage-and-list-services-in-linux/#:~:text=Managing%20Linux%20Services-,Linux%20Services,operation%20of%20the%20operating%20system.). Again, this is a testament to the fact that K3s isn't supposed to be a local development tool but rather something to use in production. This does mean that performance is great, with the [microservice demo](https://github.com/GoogleCloudPlatform/microservices-demo.git) being deployed in just about a minute.
+
+### Kubeadm
+
+![Kubeadm logo]({{site.images}}{{page.slug}}/gikIAq7.png)
```
[the remark this answers is not in the document]
Comment (msali123) [reply blog/_posts/2022-06-21-k8s-dev-solutions.md:89]: Done

File: blog/_posts/2022-06-21-k8s-dev-solutions.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+
+kubeadm is another option that only runs on Linux; however, you can make it run on all Linux-based servers from Ubuntu to [Raspberry Pi](https://www.raspberrypi.org).
+
+kubeadm isn't just a tool that lets you run local Kubernetes clusters. It was developed to help spin up Kubernetes clusters in general. This means you can find a good amount of infrastructure engineers using kubeadm to administrate clusters in production.
+
+Because of this, kubeadm is the closest you will get to a production-like cluster. You can customize every single part of Kubernetes to exactly what you need, and because of this, you'll get some of the greatest performance, with the [microservice demo](https://github.com/GoogleCloudPlatform/microservices-demo.git) deployed in just under a minute.
+
+### Docker Desktop
+
+![Docker Desktop logo]({{site.images}}{{page.slug}}/kJMf8ZB.png)
```
[the remark this answers is not in the document]
Comment (msali123) [reply blog/_posts/2022-06-21-k8s-dev-solutions.md:99]: Done

File: blog/_posts/2022-06-21-k8s-dev-solutions.md [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+Docker Desktop is supported by all the major operating systems (macOS, Windows, and Linux). Today, this is the tool you're most likely to use since most engineers will already have Docker Desktop installed.
+
+Docker Desktop runs Kubernetes by spinning up a single node in a Docker container. The biggest drawback of using it is that you can *only* get a single-node cluster; however, for most, this isn't an issue.
+
+In terms of performance, Docker Desktop performs well, deploying the [microservice demo](https://github.com/GoogleCloudPlatform/microservices-demo.git) in about a minute.
+
+### MicroK8s
+
+![MicroK8s logo]({{site.images}}{{page.slug}}/kbvMYid.png)
```
[the remark this answers is not in the document]
Comment (msali123) [reply blog/_posts/2022-06-21-k8s-dev-solutions.md:109]: Done

**Comment by msali123** [comment]
Done

**Comment by jalletto** [comment]
@msali123 Looks like the build failed again do to a linting error.

BTW I set it up so that hopefully on your next pr the build will run right away without me needing to approve it. So you'll be able to see if it fails after you push instead of having to wait for me to run it.

**Comment by adamgordonbell** [comment]
> @msali123 Looks like the build failed again do to a linting error.

I think it should be possible to run the linting in the code spaces as well. To find out faster if any problems are occurring. Just run `lint` or related commands.

**Review by jalletto — APPROVED** [review]
Looks great! Nice work.

**Pull request merged**